### PR TITLE
Chore: Remove `same-origin` dependency.

### LINKED ISSUE #879: Remove package `same-origin` with new method in `misc`
We are using the package `same-origin` somewhere in the code. It's a bit outdated and the new WHATWG URL parser available in node 8 makes things a lot easier to verify if two urls have the same origin.

We need to:

1. Create a new method under `utils/misc`: `sameOrigin` with something similar to the following:

```typescript
export const sameOrigin(url1: string, url2: string): boolean {
  return (new URL(url1)).origin === (new URL(url2)).origin;
}
```

2. Remove `same-origin` from the dependencies and use this method.

[Original conversation](https://github.com/sonarwhal/sonarwhal/pull/875#discussion_r173954925)

### PR DESCRIPTION
**Pull request by sarvaje**
Fix #879

## Pull request checklist

Make sure you:

- [ ] Signed the [Contributor License Agreement](https://cla.js.foundation/sonarwhal/sonarwhal)
- [ ] Followed the [commit message guidelines](https://sonarwhal.com/docs/contributor-guide/getting-started/pull-requests/#commit-messagess)

For non-trivial changes, please make sure you also:

- [ ] Added/Updated related documentation.
- [ ] Added/Updated related tests.

### CODE REVIEW

**Comment by alrra** [comment]
Will update `yarn.lock` before merging.

File: packages/sonarwhal/src/lib/utils/misc.ts [hunk 1]
```diff
@@ -344,6 +344,10 @@ const getSonarwhalPackage = () => {
     return require(path.join(__dirname, '../../../../package.json'));
 };
 
+const sameOrigin = (url1: string, url2: string): boolean => {
```
Comment (molant) [inline packages/sonarwhal/src/lib/utils/misc.ts]: I actually thought we were using it in more places. Maybe it wasn't worth it to add a new method to `misc`?

Comment (alrra) [reply packages/sonarwhal/src/lib/utils/misc.ts]: I can modify things.

Comment (alrra) [reply packages/sonarwhal/src/lib/utils/misc.ts]: https://github.com/sonarwhal/sonarwhal/commit/f8845722ea6b2e06578a6606590d927859abf6a9

**Pull request closed without merging**